### ISSUE TITLE
Post Date Time Zone

### ISSUE THREAD

**Issue by jdhitsolutions** [OpenLiveWriter/OpenLiveWriter#262]
I assumed the Post Date would be the me time zone, or at least what is configured for my blog but that doesn't seem to be the case. In OLW I set a time (11:50AM) but when I posted the draft to my blog it was changed to GMT time (16:50). At least that is what I am assuming. I am on Eastern time and that is how my Wordpress blog is configured. 

Not sure if I am not understanding this feature or if it is a bug.

**Comment by ScottIsAFool** [comment]
Oh, this is happening on a WordPress based blog?

**Comment by jdhitsolutions** [comment]
Yes. Is there something I'm missing? I checked the blog general settings and it is set to NY time. My local computer is set to Eastern Standard Time.

**Comment by ScottIsAFool** [comment]
Was just checking as it's a known issue with blogger, but should work fine with WordPress.

**Comment by Happy4OLW** [comment]
Sorry, didn't see this when I posted the same issue.  Time zone is EST but scheduled post for 8pm actually posts for the following day at 1am. so 5 hours later.

**Comment by willduff** [comment]
@jdhitsolutions Interesting, we've only seen this issue on Google Blogger blogs previously. I'll keep this bug open to investigate why a Wordpress blog would have this issue. Can you tell me what version of Wordpress you're using? Is this on Wordpress.com or self-hosted?

**Comment by jdhitsolutions** [comment]
It is a self-hosted site running v4.4.

**Comment by jdhitsolutions** [comment]
Just did another test. Posted a draft to my blog. OLW has a date time of 1/9/2016 12:00PM. When I edit the draft in Wordpress I see this: Schedule for: Jan 9, 2016 @ 17:00. My blog is set to the New York time zone.

**Comment by ScottIsAFool** [comment]
Just to confirm, is the server running the blog set to the right timezone?

**Comment by jdhitsolutions** [comment]
I am self-hosting Wordpress but it is on a shared server from HostGator. The server itself is on CST. But I don't think that is related. Right now I am 5 hours behind GMT. So 12:00PM would be 17:00 which is what I'm seeing. The OLW log file is using my local time if that helps.

**Comment by jdhitsolutions** [comment]
I left the date and time alone in the draft that I posted to my blog. I then brought it back down into OLW and it shows a published data of 1/9/2016 10:00PM. Now I'm really confused.

**Comment by robertjm** [comment]
I experienced this in Blogger too. After noticing it, I tried to roll the time back by eight hours. However, while it now sows the correct time, it still lists it as scheduled over in Blogger. I have to copy my source, and then post that for it to actually show up in the blog.

**Comment by ScottIsAFool** [comment]
@robertjm yep, on blogger it's a known issue right now for us,

**Comment by DawnL43072** [comment]
sigh.... will be nice when it gets fixed....thanks in advance :)

**Comment by demortes** [comment]
I was looking into this. The servers timezone from shell is accurate. The WordPress user however was defaulted to GMT. Once I "published" the scheduled post, changed the time zone, the published post schedule did not change. Updating the post from OLW however did fix the problem on the server.

Please confirm, if you go to your WordPress administration panel Settings -> General -> Timezone and set that appropriately, if you then schedule another post, what happens?

**Comment by DawnL43072** [comment]
Everything is working fine now….thanks!

**Comment by dougrathbone** [comment]
Fix submitted in #1077 — fixes backwards timezone conversion in GeneratePostDateFields. Includes 3 unit tests.
